src/popup/router/index.ts

```diff
@@ -130,12 +127,10 @@ router.beforeEach(async (to, from, next) => {
 });
 
 router.afterEach(async (to) => {
-  if (RUNNING_IN_POPUP) return;
-  if (to.meta?.notPersist) {
-    await browser?.storage.local.remove(lastRouteKey);
-  } else {
-    await browser?.storage.local.set({ [lastRouteKey]: to.path });
+  if (RUNNING_IN_POPUP) {
+    return;
   }
+  setLastPath(to.meta?.notPersist ? '' : to.path);
```

Comment:
```suggestion
  if (RUNNING_IN_POPUP) {
    setLastPath(to.meta?.notPersist ? '' : to.path);
  }
```

But to be honest I would remove the test because I don't see any downside of seving the last used path to local storage on every environment. The matter is what we do with this value.

Reply:
Not awaiting for a `await browser.storage.local.get` and `await browser?.storage.local.set/remove` methods brought us an optimization of ~200ms on popup opening.
https://github.com/aeternity/superhero-wallet/pull/2077#issue-1709559390

Reply:
@CedrikNikita yes, your change brought optimisation because there was `await` there. In Paweł's implementation there is no await.